### PR TITLE
Feature/customers methods

### LINKED ISSUE #138: Add customers methods in docs

### PR DESCRIPTION
**Pull request by jaepass**
Added customers endpoint methods, edited order of docs and minor typos

Fixes #138

### CODE REVIEW

File: docs/04_Customers.md [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+customers endpoint will return the full customers object with data to create a customer's authorized view containing
+information such as their contact name and list of orders.
+
+---
+
+
+## Send login token
+
+This endpoint helps to authenticate and resolve a customer in the merchant account by accepting an email address as an
+argument and send a one-click login email. The email contains a short-lived magic link for the user to access a
```
Comment (robbieaverill) [inline docs/04_Customers.md:22]:
```suggestion
argument and sending a one-click login email. The email contains a short-lived magic link for the user to access a
```

File: docs/04_Customers.md [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+particular customer granting access to that customer's orders, addresses, etc.
+
+Example request using Commerce.js:
+
+```js
+import Commerce from '@chec/commerce.js';
+
+const commerce = new Commerce('{your_public_key}');
+
+commerce.customers.login('leslie.lawless@example.com', 'commodi').then((token) => console.log(token));
```
Comment (robbieaverill) [inline docs/04_Customers.md:35]:
```suggestion
commerce.customers.login('leslie.lawless@example.com', 'https://yourwebsite.com/login/callback').then((token) => console.log(token));
```

File: docs/04_Customers.md [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+```
+
+Example request using cURL:
+
+```bash
+curl -X POST \
+    "https://api.chec.io/v1/customers/email-token" \
+    -H "Content-Type: application/json" \
+    -d '{"email":"leslie.lawless@example.com","base_url":"commodi"}'
```
Comment (robbieaverill) [inline docs/04_Customers.md:44]:
```suggestion
    -d '{"email":"leslie.lawless@example.com","base_url":"https://yourwebsite.com/login/callback"}'
```

File: docs/04_Customers.md [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+</div>
+
+When this endpoint is called in a login flow, the subsequent method below to call would exchange the token for a JSON
+web token to log the customer in to a protected route.
+
+---
+
+## Get access token
+
+The `getToken()` method uses `POST /v1/customers/exchange-token` to obtain a JSON web token scoped to the customer the
```
Comment (robbieaverill) [inline docs/04_Customers.md:63]:
```suggestion
Once a customer has entered their email address in the previous step and clicked on the one-time login link they received in their email, the link will send them back to your server (see https://yourwebsite.com/login/callback in the example above). In that callback handler, you can use the "get access token" method to exchange the token in the URL for a JSON web token.

The `getToken()` method uses `POST /v1/customers/exchange-token` to obtain a JSON web token scoped to the customer the
```

File: docs/04_Customers.md [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+    -d '{"token":"1ae420a5-2f43-426f-8a61-516eabb56d33"}'
+```
+
+| Method | Description |
+| -------------------- | ----------- |
+| `getToken(token, save = true)`  | Get access token |
+
+<div class="highlight highlight--info">
+    <span>Info</span>
+    <p>If you want to save the customer in local session storage to persist the data, pass in the optional <code>save = true</code> argument.</p>
```
Comment (robbieaverill) [inline docs/04_Customers.md:94]:
```suggestion
    <p>If you do not want to save the customer in local session storage, pass false for the second argument <code>save</code>.</p>
```

File: docs/04_Customers.md [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+const commerce = new Commerce('{your_public_key}');
+
+commerce.customers.update({
+        "email":"leslie.lawless@example.com",
+        "firstname":"Leslie",
+        "lastname":"Lawless",
+        "external_id":"MY_CRM_USER_123"
+    },
+    'cstmr_K1YDR2qy29Qem6'
+).then((customer) => console.log(customer));
```
Comment (robbieaverill) [inline docs/04_Customers.md:123]:
```suggestion
commerce.customers.update({
  email: 'leslie.lawless@example.com',
  firstname: 'Leslie',
  lastname: 'Lawless',
  external_id: 'MY_CRM_USER_123',
}, 'cstmr_K1YDR2qy29Qem6')
  .then((customer) => console.log(customer));
```

File: docs/04_Customers.md [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+order](/docs/api/?shell#retrieve-order).
+
+Example request using Commerce.js:
+
+```js
+import Commerce from '@chec/commerce.js';
+
+const commerce = new Commerce('{your_public_key}');
+
+commerce.customers.getOrders('cstmr_f89398fs489g').then((order) => console.log(order));
```
Comment (robbieaverill) [inline docs/04_Customers.md:204]:
```suggestion
commerce.customers.getOrders('ord_Kvg9l6zvnl1bB7').then((order) => console.log(order));
```

File: docs/04_Customers.md [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+```
+
+Example request using cURL:
+
+```bash
+curl -X GET \
+    -G "https://api.chec.io/v1/customers/cstmr_f89398fs489g/orders/ord_Kvg9l6zvnl1bB7" \
+    -H "X-Authorization: {token}"
```
Comment (robbieaverill) [inline docs/04_Customers.md:212]:
```suggestion
    -H "Authorization: Bearer {token}"
```

File: docs/04_Customers.md [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+
+commerce.customers.getOrders('cstmr_K1YDR2qy29Qem6').then((orders) => console.log(orders));
+```
+
+Example request using cURL:
+
+```bash
+curl -X GET \
+    -G "https://api.chec.io/v1/customers/cstmr_f89398fs489g/orders" \
+    -H "X-Authorization: {token}"
```
Comment (robbieaverill) [inline docs/04_Customers.md:172]:
```suggestion
    -H "Authorization: Bearer {token}"
```

File: docs/04_Customers.md [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+    'cstmr_K1YDR2qy29Qem6'
+).then((customer) => console.log(customer));
+```
+
+Example request using cURL:
+
+```bash
+curl -X PUT \
+    "https://api.chec.io/v1/customers/cstmr_K1YDR2qy29Qem6" \
+    -H "X-Authorization: {token}" \
```
Comment (robbieaverill) [inline docs/04_Customers.md:131]:
```suggestion
    -H "Authorization: Bearer {token}" \
```

File: docs/04_Customers.md [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+customer by it's ID.
+
+Example request using Commerce.js:
+
+```js
+import Commerce from '@chec/commerce.js';
+
+const commerce = new Commerce('{your_public_key}');
+
+commerce.customers.about('cstmr_K1YDR2qy29Qem6').then((customer) => console.log(customer));
```
Comment (robbieaverill) [inline docs/04_Customers.md:243]:
```suggestion
commerce.customers.about().then((customer) => console.log(customer));
```

File: docs/04_Customers.md [hunk 12]
```diff
@@ -0,0 +1,343 @@
...
+```
+
+Example request using cURL:
+
+```bash
+curl -X GET \
+    -G "https://api.chec.io/v1/customers/cstmr_K1YDR2qy29Qem6" \
+    -H "X-Authorization: {token}"
```
Comment (robbieaverill) [inline docs/04_Customers.md:251]:
```suggestion
    -H "Authorization: Bearer {token}"
```

File: docs/04_Customers.md [hunk 13]
```diff
@@ -0,0 +1,343 @@
...
+```
+
+| Method | Description |
+| -------------------- | ----------- |
+| `about()`  | Gets information about the currently authorized customer |
+
+<div class="highlight highlight--note">
+    <span>Note</span>
+    <p>Refer to the full response for retrieving a customer by it's ID <a href="/docs/api/?shell#get-customer">here</a>.</p>
```
Comment (robbieaverill) [inline docs/04_Customers.md:260]:
```suggestion
    <p>Refer to the full response for retrieving a customer by its ID <a href="/docs/api/?shell#get-customer">here</a>.</p>
```

File: docs/04_Customers.md [hunk 14]
```diff
@@ -0,0 +1,343 @@
...
+    <span>Note</span>
+    <p>Refer to the full response for retrieving an order for a customer <a href="/docs/api/?shell#get-order-for-customer">here</a>.</p>
+</div>
+
+---
+
+## Get customer
+
+The `about()` method uses `GET /v1/customers/{customer_id}` to return information about the currently authorized
+customer by it's ID.
```
Comment (robbieaverill) [inline docs/04_Customers.md:234]:
```suggestion
customer by its ID.
```

File: docs/04_Customers.md [hunk 15]
```diff
@@ -0,0 +1,343 @@
...
+The `id()` method returns the customer ID used during a successful login from local storage.
+
+Example request using Commerce.js:
+
+```js
+import Commerce from '@chec/commerce.js';
+
+const commerce = new Commerce('{your_public_key}');
+
+commerce.customers.id().then((customerId) => console.log(customerId));
```
Comment (robbieaverill) [inline docs/04_Customers.md:276]:
```suggestion
console.log(commerce.customers.id());
```

File: docs/04_Customers.md [hunk 16]
```diff
@@ -0,0 +1,343 @@
...
+
+| Method | Description |
+| -------------------- | ----------- |
+| `id()`  | Gets the customer ID used during a successful login |
+
+---
+
+## Get customer token
+
+The `token()` method returns the customer token used during a successful login from local storage.
```
Comment (robbieaverill) [inline docs/04_Customers.md:287]:
```suggestion
The `token()` method returns the customer's JSON web token from local storage, if logged in.
```

File: docs/04_Customers.md [hunk 17]
```diff
@@ -0,0 +1,343 @@
...
+
+Example request using Commerce.js:
+
+```js
+import Commerce from '@chec/commerce.js';
+
+const commerce = new Commerce('{your_public_key}');
+
+commerce.customers.token().then((token) => console.log(token));
```
Comment (robbieaverill) [inline docs/04_Customers.md:296]:
```suggestion
console.log(commerce.customers.token());
```

File: docs/04_Customers.md [hunk 18]
```diff
@@ -0,0 +1,343 @@
...
+```
+
+| Method | Description |
+| -------------------- | ----------- |
+| `token()`  | Gets the customer token used during a successful login |
```
Comment (robbieaverill) [inline docs/04_Customers.md:301]:
```suggestion
| `token()`  | Gets the customer's JSON web token, if logged in |
```

File: docs/04_Customers.md [hunk 19]
```diff
@@ -0,0 +1,343 @@
...
+The `isLoggedIn()` method checks if a customer is logged in the current session and returns either `true` or `false`.
+
+Example request using Commerce.js:
+
+```js
+import Commerce from '@chec/commerce.js';
+
+const commerce = new Commerce('{your_public_key}');
+
+commerce.customers.isLoggedIn().then((response) => console.log(response));
```
Comment (robbieaverill) [inline docs/04_Customers.md:316]:
```suggestion
console.log(commerce.customers.isLoggedIn());
```

File: docs/04_Customers.md [hunk 20]
```diff
@@ -0,0 +1,343 @@
...
+The `logout()` method logs the customer out by clearing the customer state from the current session.
+
+Example request using Commerce.js:
+
+```js
+import Commerce from '@chec/commerce.js';
+
+const commerce = new Commerce('{your_public_key}');
+
+commerce.customers.logout().then((response) => console.log(response));
```
Comment (robbieaverill) [inline docs/04_Customers.md:336]:
```suggestion
commerce.customers.logout();
```

**Review by robbieaverill — CHANGES_REQUESTED** [review]
Nice work! The feedback I've left is mostly architectural, there's a couple of typos in there too.

**Comment by jaepass** [comment]
Awesome thank you so much for review! I'll apply those changes stat!

File: docs/04_Customers.md [hunk 21]
```diff
@@ -0,0 +1,344 @@
...
+order](/docs/api/?shell#retrieve-order).
+
+Example request using Commerce.js:
+
+```js
+import Commerce from '@chec/commerce.js';
+
+const commerce = new Commerce('{your_public_key}');
+
+commerce.customers.getOrders('ord_Kvg9l6zvnl1bB7').then((order) => console.log(order));
```
Comment (robbieaverill) [inline docs/04_Customers.md:205]:
```suggestion
commerce.customers.getOrder('ord_Kvg9l6zvnl1bB7').then((order) => console.log(order));
```

**Review by robbieaverill — COMMENTED** [review]
One last thing!

File: docs/04_Customers.md (hunk 21 above)
Comment (jaepass) [reply docs/04_Customers.md:205]: Oops sorry missed this, thanks!

**Review by robbieaverill — APPROVED** [review]

**Pull request merged**